lib/collection/src/update_handler.rs

```diff
@@ -285,18 +335,27 @@ impl UpdateHandler {
         optimization_handles: Arc<TokioMutex<Vec<StoppableTaskHandle<bool>>>>,
         max_handles: usize,
     ) {
-        while let Some(signal) = receiver.recv().await {
-            match signal {
-                OptimizerSignal::Nop | OptimizerSignal::Operation(_) => {
+        loop {
+            let receiver = timeout(OPTIMIZER_CLEANUP_INTERVAL, receiver.recv());
+            let result = receiver.await;
+
+            // Always clean up on any signal
+            Self::cleanup_optimization_handles(optimization_handles.clone()).await;
```

Comment:
Pointer: the worker now waits for any signal, but also continues the loop every 5 seconds.

This way we always clean up finished optimization tasks at least every 5 seconds, as that is done right after. This is needed to catch and propagate panics in time.

If we actually received a signal, it is handled below.